crates/hir/src/lib.rs

```diff
@@ -3472,13 +3682,8 @@ impl Impl {
                     .filter(filter),
             )
         });
-        for id in def_crates
-            .iter()
-            .flat_map(|&id| Crate { id }.transitive_reverse_dependencies(db))
-            .map(|Crate { id }| id)
-            .chain(def_crates.iter().copied())
-            .unique()
-        {
```

Comment:
This change seems odd. We don't need to traverse all crates in the crate graph, only the ones that can observe the type no?

Reply:
Hmm, works if I chain the current crate to it. I guess I'll do that then